SendWithUs.Client/Requests/SendRequest.cs

```diff
@@ -22,9 +22,7 @@ namespace SendWithUs.Client
 {
     using System;
     using System.Collections.Generic;
-    using System.Linq;
     using Newtonsoft.Json;
-    using Newtonsoft.Json.Linq;
 
     /// <summary>
     /// Represents the data necessary to make API requests to send email messages.

```

Comment:
The constraints around the Data property in the generic version (`SendRequest<TData>`) seem a little lose and may be a cause for it not being used as much as the base `SendRequest` type. In the past I have handled this by having for example, `SendRequest<TData>` inherit from `ISendRequest` directly while explicitly implementing the `object ISendRequest.Data` property. I could then have the "non-generic" `SendRequest` inherit from `SendRequest<object>` . After such a change it may even be found that the usages of `SendRequest` instances could made obsolete in favor of the generic version to preserve the type of `Data`. This would come at a cost however of having to explicitly specify the generic type parameter when using the constructor or using a method that could provide type inference, like `SendRequest.Create(id, address, data)`.